MetadataExtractor/Formats/Gif/GifReader.cs

```diff
@@ -332,17 +333,21 @@ private static GifImageDirectory ReadImageBlock(SequentialReader reader)
         private static byte[] GatherBytes(SequentialReader reader)
         {
             var bytes = new MemoryStream();
-            var buffer = new byte[257];
+            var buffer = ArrayPool<byte>.Shared.Rent(257);
 
             while (true)
             {
-                var b = reader.GetByte();
-                if (b == 0)
-                    return bytes.ToArray();
-                buffer[0] = b;
-                reader.GetBytes(buffer, 1, b);
-                bytes.Write(buffer, 0, b + 1);
+                var len = reader.GetByte();
+                if (len == 0)
+                    break;
+                buffer[0] = len;
+                reader.GetBytes(buffer, offset: 1, count: len);
+                bytes.Write(buffer, 0, len + 1);
             }
+
+            ArrayPool<byte>.Shared.Return(buffer);
```

Comment:
In any case where there's the possibility of throwing (e.g. calling an underlying stream we don't control), we should consider wrapping that logic try + finally.

```csharp
try
{ 
   // logic that may throw
}
finally
{
  ArrayPool<byte>.Shared.Return(buffer);
}
```

To ensure we don't leak a buffer and degrade the performance of the ArrayPool for other callers.

This should be unnecessary when we're confident we won't throw.

Reply:
I added try/finally in a few places but not in others. Leaking a buffer isn't the end of the world. There can be a perf hit for having try/catch, even when nothing actually throws at run-time.